resin-io-modules/resin-procbots
# 374 log redaction

* SyncBot: Log the full error response from services.
* ProcBot: Add function to calculate the key/value pairs that would be injected.
* ServiceScaffold: Store logger instance and construction data when constructed.
* ServiceScaffold: Bubble to error handler details of the request and the session.
* Logger: Allow configuration of a list of strings to redact.
* Logger: Move log redaction to the underlying, universal, function to catch more outputs.

Test routine:
--

1) For each of Discourse, Front & Flowdock:
    1) Introduce a deliberate error in the payload delivered to the service
    1) Capture an output of that error occurring on the master branch
    1) Capture an output of that error occurring on the feature branch
    1) Confirm that I can see the error in the second capture
    1) Confirm that secrets are not present in the second capture
1) Remove the captures via their own commit, so they can be cited for now, but disappear on merge

Change-Type: patch
Connects-To: #374
See: https://github.com/resin-io-modules/resin-procbots/commit/a915cbc983d089e3d7ddcabb9ff890859cc37e86

---
##### Contributor checklist

- [x] Build output been rebuilt and tested
- [ ] Introduces security considerations
- [ ] Documentation is added or changed
- [ ] Affects the development, build or deployment processes of the component

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---
---- Autogenerated Waffleboard Connection: Connects to #374
